bioimageio/spec/commands.py

```diff
@@ -120,3 +160,67 @@ def validate(
         "traceback": tb,
         "warnings": ValidationWarning.get_warning_summary(validation_warnings),
     }
+
+
+def update_rdf(
+    source: Union[RawResourceDescription, dict, os.PathLike, IO, str, bytes],
+    update: Union[RawResourceDescription, dict, os.PathLike, IO, str, bytes],
+    output: Union[dict, os.PathLike] = None,
+    validate_output: bool = True,
+) -> Union[dict, Path, RawResourceDescription]:
+    """
+    Args:
+        source:  source of RDF
+        update:  a (partial) RDF used as update
+        output:  dict or path to write output to (default: return new dict)
+        validate_output: whether or not to validate the updated RDF
+
+    Returns:
+        The updated content of the source rdf as dict or,
+        if output is a path, that path (where the updated content is saved to).
+
+    Raises:
+        ValidationError: if `validate_output` and the updated rdf does not pass validation
+    """
```

Comment:
very close to the initial draft, I only added `RawResourceDescription` for source and update, so one does not have to go back to dicts if the RDF(s) are already loaded.